src/components/progress-circle/progress_circle.ts

```diff
@@ -0,0 +1,102 @@
+import {Component, ViewEncapsulation, ElementRef, Attribute} from 'angular2/core';
+import {isPresent, CONST} from 'angular2/src/facade/lang';
+
+
+// TODO(josephperrott): Benchpress tests.
+
+/** Display modes of Progress Circle */
+@CONST()
+class ProgressMode {
+  @CONST() static DETERMINATE = 'determinate';
+  @CONST() static INDETERMINATE = 'indeterminate';
+}
+
+
+/**
+ * <md-progress-circle> component.
+ */
+@Component({
+  selector: 'md-progress-circle',
+  inputs: ['value'],
+  host: {
+    '[attr.mode]': 'mode',
+    'role': 'progressbar',
+    'aria-valuemin': '0',
+    'aria-valuemax': '100',
+    '[attr.aria-valuenow]': 'value'
+  },
+  templateUrl: './components/progress-circle/progress_circle.html',
+  styleUrls: ['./components/progress-circle/progress-circle.css'],
+  encapsulation: ViewEncapsulation.None
+})
+export class MdProgressCircle {
+  /** Internal value for the progress circle to be used when in determinate mode. */
+  value_: number = 0;
+  /** Mode of the progress circle */
+  mode: string;
+
+
+  constructor(@Attribute('mode') mode: string) {
+    /**
+     * Checks if the mode provided is a valid mode.  If it is not a valid mode, sets
+     * the mode to determinate.
+     */
+    this.mode = (mode == ProgressMode.INDETERMINATE ||
+                 mode == ProgressMode.DETERMINATE) ? mode : ProgressMode.DETERMINATE;
+  }
+
+
+  /** Getter for the current strokeOffset to represent the progress circle. */
```

Comment:
Could you add something to give a basic understanding of what the stroke offset _is_?